### ISSUE TITLE
sort LicenseFinder::MergedPackage failed

### ISSUE THREAD

**Issue by guyromb** [pivotal/LicenseFinder#223]
I get the following error, when running with a report generation (works when just listing STDOUT).

`C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/lib/license_finder/report.rb:16:in 'sort': comparison of LicenseFinder::MergedPackage with LicenseFinder::MergedPackage failed (ArgumentError)
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/lib/license_finder/report.rb:16:in 'sorted_dependencies'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/lib/license_finder/reports/csv_report.rb:16:in 'block in to_s'
        from C:/Ruby23-x64/lib/ruby/2.3.0/csv.rb:1166:in 'generate'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/lib/license_finder/reports/csv_report.rb:15:in 'to_s'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:79:in 'say'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/shell.rb:59:in 'say'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/lib/license_finder/cli/main.rb:76:in 'report'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/command.rb:27:in 'run'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in 'invoke_command'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor.rb:359:in 'dispatch'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/base.rb:440:in 'start'
        from C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/license_finder-2.1.0/bin/license_finder:5:in '<top (required)>'
        from C:/Ruby23-x64/bin/license_finder:23:in 'load'
        from C:/Ruby23-x64/bin/license_finder:23:in '<main>'`

**Comment by flavorjones** [comment]
Hi,

Thanks for opening this issue.

Can you help me reproduce what you're seeing? Would it be possible to provide a sample project?

**Comment by josearmandojacq** [comment]
The same is happening to me. When i do just `license_finder` inside a project I can see the dependencies, but when trying to save a report `license_finder --format=html --save="some path" --decisions-file="some path"`. It shows: `comparison of LicenseFinder::MergedPackage with LicenseFinder::MergedPackage failed (ArgumentError)`.
The interesting bit is that it appeared after running it a couple of times.

**Comment by jacob-ewald** [comment]
I ran into this same problem with the current version (5.9.2). I could reproduce it consistently by manually adding a dependency via:
```
license_finder dependencies add ZipArchive MIT --homepage="https://github.com/ZipArchive/ZipArchive" --approve
```
What fixed it, for me, was to specify a version number.
```
license_finder dependencies add ZipArchive MIT 2.2.2 --homepage="https://github.com/ZipArchive/ZipArchive" --approve
```
The [help text](https://github.com/pivotal/LicenseFinder/blob/master/lib/license_finder/cli/dependencies.rb#L12) specifies that the `VERSION` parameter is optional, and it even defaults to `nil` in the [method signature](https://github.com/pivotal/LicenseFinder/blob/master/lib/license_finder/cli/dependencies.rb#L13), however, it's expected to have _some_ value when sorting for the report.
It might be a similar issue to #313.

**Comment by andrejska** [comment]
Yes, @jacob-ewald has correctly described this issue, repeats in latest version as well (5.10.2)

**Comment by josselinchevalay** [comment]
hello guys same issue with `--homepage` flag when i forget that that's works fine.

**Comment by kdykeman** [comment]
Hi @jacob-ewald, thanks for making this easy to reproduce. The behavior definitely doesn't match the description (and it shouldn't error out), but it's not clear to me yet which behavior should be correct.
1. The `dependencies add` feature is meant to "Add or remove dependencies that your package managers are not aware of". So in this case, you'd be adding a dependency without specifying a version (i.e. a dependency with no version). Is that the functionality you want? In other words, should LF allow you declare a dependency on an unversioned library?
1. The `approvals add` feature doesn't currently support version but should soon support that as optional with the idea that you may want to approve a specific version of a dependency. Is that more what you're after?

Any additional information about your use case would definitely be appreciated.

**Comment by jacob-ewald** [comment]
I think making the `VERSION` parameter required when doing a `dependency add` would alleviate the problem, which, I believe, lines up with your first point. At the time of adding the dependency, it definitely has _some_ sort of version number, even if it's something like `0.0.1`. 

When I ran into this I was (mistakenly) thinking that because I specified the `--homepage` that there was logic in LF that would grab the current version number at that time from the homepage. I then left off the `VERSION` parameter out of laziness (and a faulty assumption/hope).
